changecalc/main.go

```diff
@@ -0,0 +1,276 @@
+	for _, item := range config.ResetList {
+		resetMap[item] = empty
+	}
+
+	return ParsedConfig{Config: config, ResetMap: resetMap}
+}
+
+// Returns true if this directory belongs to the current go module (identified by MODULE_NAME_SOURCE file)
+func checkIfBelongs(dir string, cache map[string]bool, rootIdName string) bool {
+	if dir == "." || dir == "" {
+		return true
+	}
+
+	val, ok := cache[dir]
+	if ok {
+		return val
+	}
+
+	result := false
+
+	//fmt.Fprintf(os.Stderr, "Checking %s for go.mod\n", dir)
+	_, err := os.Stat(filepath.Join(dir, rootIdName))
+	if err != nil {
+		// assuming it's not-found
+		result = checkIfBelongs(filepath.Dir(dir), cache, rootIdName)
+	}
+	// If we found go.mod, it's a root of another module.
+	// "go test" will fail if we pass this directory:
+	// % go test ./bundle/internal/tf/codegen/schema
+	// main module (github.com/databricks/cli) does not contain package github.com/databricks/cli/bundle/internal/tf/codegen/schema
+	cache[dir] = result
+	return result
+}
+
+// readModuleName parses the go.mod file to extract the module name.
+func readModuleName(goModPath string) (string, error) {
+	data, err := os.ReadFile(goModPath)
+	if err != nil {
+		return "", err
+	}
+	modFile, err := modfile.Parse(goModPath, data, nil)
+	if err != nil {
+		return "", err
+	}
+	return modFile.Module.Mod.Path, nil
+}
+
+// resolvePackages converts file paths to fully qualified package names.
+// Recursively find all packages that depend on the given list
+func findDependents(initialPackages iter.Seq[string], reverseDeps map[string][]string) map[string]struct{} {
+	visited := make(map[string]struct{})
+	var visit func(pkg string)
+	visit = func(pkg string) {
+		if _, seen := visited[pkg]; seen {
+			return
+		}
+		visited[pkg] = struct{}{}
+		for _, dependent := range reverseDeps[pkg] {
+			visit(dependent)
+		}
+	}
+	for pkg := range initialPackages {
+		visit(pkg)
+	}
+	return visited
+}
+
+// findNearestGoPackage returns the closest parent (or itself) directory
+// that contains at least one *.go file. If no such directory is found,
+// it returns an empty string.
+func findNearestGoPackage(directory string) string {
+	for {
+		if hasGoFiles(directory) {
+			return directory
+		}
+
+		parent := filepath.Dir(directory)
+
+		if parent == directory {
+			break
+		}
+
+		directory = parent
+	}
+
+	return ""
+}
+
+// hasGoFiles checks if a directory contains any .go files
+func hasGoFiles(directory string) bool {
+	files, err := os.ReadDir(directory)
+	if err != nil {
+		return false
+	}
+
+	for _, file := range files {
+		if !file.IsDir() && strings.HasSuffix(file.Name(), ".go") {
+			return true
+		}
+	}
+	return false
+}
+
+// GetChangedFiles compares the current branch to the base branch
+// and returns a slice of file paths that have been modified.
+func GetChangedFiles(baseBranch string) ([]string, error) {
+	command := []string{"git", "diff", baseBranch, "--name-only", "--", "."}
```

Comment:
Should we use `git status` here if this is meant to facilitate local development?

Reply:
Not sure what you mean? Could you clarify how "git status" would be used?